src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/ValidateAnomalyDetectorTransportAction.java

```diff
@@ -0,0 +1,226 @@
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            resolveUserAndExecute(
+                user,
+                AnomalyDetector.NO_ID,
+                filterByEnabled,
+                listener,
+                () -> validateExecute(request, user, listener),
+                client,
+                clusterService,
+                xContentRegistry,
+                anomalyDetector,
+                false,
+                false
+            );
+        } catch (Exception e) {
+            logger.error(e);
+            listener.onFailure(e);
+        }
+    }
+
+    private void validateExecute(
+        ValidateAnomalyDetectorRequest request,
+        User user,
+        ActionListener<ValidateAnomalyDetectorResponse> listener
+    ) {
+        ActionListener<ValidateAnomalyDetectorResponse> validateListener = ActionListener.wrap(response -> {
+            logger.info("Result of validation process " + response);
+            // forcing response to be empty
+            listener.onResponse(new ValidateAnomalyDetectorResponse((DetectorValidationIssue) null));
+        }, exception -> {
+            if (exception instanceof ADValidationException) {
+                DetectorValidationIssue issue = parseADValidationException((ADValidationException) exception);
+                listener.onResponse(new ValidateAnomalyDetectorResponse(issue));
+                return;
+            }
+            listener.onFailure(exception);
+        });
+        ValidateAnomalyDetectorActionHandler handler = new ValidateAnomalyDetectorActionHandler(
+            clusterService,
+            client,
+            validateListener,
+            anomalyDetectionIndices,
+            request.getDetector(),
+            request.getRequestTimeout(),
+            request.getMaxSingleEntityAnomalyDetectors(),
+            request.getMaxMultiEntityAnomalyDetectors(),
+            request.getMaxAnomalyFeatures(),
+            RestRequest.Method.POST,
+            xContentRegistry,
+            user,
+            searchFeatureDao,
+            request.getTypeStr()
+        );
+        try {
+            handler.start();
+        } catch (Exception exception) {
+            String errorMessage = String
+                .format(Locale.ROOT, "Unknown exception caught while validating detector %s", request.getDetector());
+            logger.error(errorMessage, exception);
+            listener.onFailure(exception);
+        }
+    }
+
+    protected DetectorValidationIssue parseADValidationException(ADValidationException exception) {
+        String originalErrorMessage = exception.getMessage();
+        String errorMessage;
+        Map<String, String> subIssues = null;
+        Object suggestion = null;
+        switch (exception.getType()) {
+            case FEATURE_ATTRIBUTES:
+                logger.info("originalErrorMessage: " + originalErrorMessage);
```

Comment:
> The message is from AbstractAnomalyDetectorActionHandler.validateAnomalyDetectorFeatures right?

Correct.

For now I may not want to show it directly to users because the nested exception might have unexpected long message like the example you shared. On current UI, validation error message is directly shown to users, and once validate API is integrated, error message will come from validation response. Very long message may not look good. 

I think we can start from current solution, and see feedback from users. If current message is not helpful enough for user, we may need to think about a good way, both front-end/back-end to show nested details in friendly way to user, even such detailed message is very long.